core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer_utils.js

```diff
@@ -0,0 +1,139 @@
+const puppeteer = require('puppeteer');
+const testConstants = require('./testConstants.js');
+
+module.exports = class puppeteerUtilities {
+  page;
+  browserObject;
+  isCookieAccepted = false;
+
+  /**
+   * This is a function that opens a new browser instance for the user.
+   * @returns {Promise<puppeteer.Page>} - Returns a promise that resolves to a Page object controlled by Puppeteer.
+   */
+  async openBrowser(){
+   /* currently, headless is set to false and the page viewport
+       is maximized so that it would be easy for the developers
+       to debug easily while testing.
+       We can remove these settings before merging as we have
+       to run the tests in headless mode. */
+    await puppeteer
+      .launch({
+        headless: false,
+        args: ['--start-fullscreen', '--use-fake-ui-for-media-stream']
+      })
+      .then(async (browser) => {
+        this.browserObject = browser;
+        this.page = await browser.newPage();
+        await (this.page).setViewport({ width: 0, height: 0 });
+        await this.page.on('dialog', async dialog => {  // accepting the alerts that appear in between the tests
```

Comment:
Put comments above the line that they pertain to, don't exceed the line length limit.